### PR TITLE
feat(ckb-indexer): can get cell with block hash

### PR DESCRIPTION
**Pull request by sighWang**
* copy plimit and queue file cause es-module import error
* ignore and add todo to some ts type warning

### CODE REVIEW

File: packages/ckb-indexer/src/pLimit.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import Queue from "./queue";
```
Comment (homura) [inline packages/ckb-indexer/src/pLimit.ts]: to make the source trackable, maybe we should move the file to `ckb-index/src/third-party/*`, and add a comment about the source's GitHub URL

**Comment by homura** [comment]
we should add comments to `CKBIndexer` and `CKBCellCollector` to describe how `withBlockHash` works, and we should also add a migration doc to describe how to migrate from `NativeIndexer` to `CKBIndexer`

File: packages/ckb-indexer/src/collector.ts [hunk 2]
```diff
@@ -234,10 +241,33 @@ export class IndexerCollector implements BaseCellCollector {
     return counter;
   }
 
+  private async getLiveCellWithBlockHash(lastCursor?: string) {
+    let result: GetCellsResults = await this.getLiveCell(lastCursor);
+    const limit = pLimit(10);
+    //TODO refine this functions type
+    const inputs = result.objects.map((cell) => {
```
Comment (homura) [inline packages/ckb-indexer/src/collector.ts]: found the JSON-RPC is [batchable](https://www.jsonrpc.org/specification#batch)，maybe we can implement via the batch spec.

Comment (sighWang) [reply packages/ckb-indexer/src/collector.ts]: use json-rpc batch and test success both local and testnet rpc server

File: packages/ckb-indexer/src/collector.ts [hunk 3]
```diff
@@ -12,13 +12,26 @@ import {
   GetCellsResults,
   HexadecimalRange,
   Order,
+  rpcResponse,
   SearchFilter,
 } from "./indexer";
 
 import { CkbIndexer, ScriptType, SearchKey } from "./indexer";
+import axios from "axios";
```
Comment (homura) [inline packages/ckb-indexer/src/collector.ts]: axios is too heavy, we can switch to cross-fetch

File: packages/ckb-indexer/src/collector.ts [hunk 4]
```diff
@@ -234,10 +245,64 @@ export class IndexerCollector implements BaseCellCollector {
...
+    }
+    const requestData = result.objects.map((cell, index) => {
+      return {
+        id: index,
+        jsonrpc: "2.0",
+        method: "get_block_hash",
+        params: [cell.block_number],
+      };
+    });
+    const blockHashList: any[] = await this.request(
```
Comment (homura) [inline packages/ckb-indexer/src/collector.ts]:
```suggestion
    // TODO define type here
    const blockHashList: any[] = await this.request(
```

File: packages/ckb-indexer/src/collector.ts (hunk 3 above)
Comment (sighWang) [reply packages/ckb-indexer/src/collector.ts]: refactor(ckb-indexer): use cross-fetch instand of axios

File: packages/ckb-indexer/src/collector.ts (hunk 4 above)
Comment (sighWang) [reply packages/ckb-indexer/src/collector.ts]: feat(ckb-indexer): add type to blockHash list

File: packages/ckb-indexer/src/collector.ts [hunk 5]
```diff
@@ -24,6 +24,12 @@ export interface OtherQueryOptions {
   ckbRpcUrl: string;
 }
 
+export interface RPCResult {
```
Comment (homura) [inline packages/ckb-indexer/src/collector.ts]:
```suggestion
interface GetBlockHashRPCResult {
```

Comment (sighWang) [reply packages/ckb-indexer/src/collector.ts]: refactor(ckb-indexer): rename

**Pull request merged**